source/extensions/http/cache/file_system_http_cache/lookup_context.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include <memory>
+
+#include "source/extensions/common/async_files/async_file_handle.h"
+#include "source/extensions/filters/http/cache/http_cache.h"
+#include "source/extensions/http/cache/file_system_http_cache/cache_file_fixed_block.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+namespace FileSystemHttpCache {
+
+class FileSystemHttpCache;
+
+using Envoy::Extensions::Common::AsyncFiles::AsyncFileHandle;
+using Envoy::Extensions::Common::AsyncFiles::CancelFunction;
+
+class FileLookupContext : public LookupContext {
+public:
+  FileLookupContext(FileSystemHttpCache& cache, LookupRequest&& lookup)
+      : cache_(cache), key_(lookup.key()), lookup_(std::move(lookup)) {}
+
+  // From LookupContext
+  void getHeaders(LookupHeadersCallback&& cb) final;
+  void getBody(const AdjustedByteRange& range, LookupBodyCallback&& cb) final;
+  void getTrailers(LookupTrailersCallback&& cb) final;
+  void onDestroy() final;
+  // This shouldn't be necessary since onDestroy is supposed to always be called, but in some
```

Comment:
Is it undefined behavior if the destructor blocks until the other thread has completed? I would expect it to be undefined behavior only when the vtable of the object is downgraded [and since there's no subclass of FileLookupContext, no changes to vtable occur until the *end* of the destructor].